File: web@bkdnoj/2020/AdvancedMath/M1/Math1_3.cpp
// http://bkdnoj.dut.udn.vn/public/practice_problem.php?id=Math1_3
#include <bits/stdc++.h>
using namespace std;

const int M = int(1e9) + 7;
long long T, n;

long long present(long long n){
	if( n == 1 ) return 1;
	if( n == 2 ) return 6;
	if( n % 2 ){
		return ( ((n-1)*6 - (n-2)) % M + (4*(n-1)/2 % M)*((n-2) %M) ) % M;
	} else {
		return ( ((n-1)*6 - (n-2)) % M + (4*(n-2)/2 % M)*((n-1) %M) ) % M;
	}
}

int main(){
	for(cin >> T; T; T--){
		cin >> n;
		printf("%lld\n", present(n));
	}
}

File: web@bkdnoj/2020/AdvancedMath/M1/Math1_3.py
def present(n):
    if n == 1:
        return 1
    if n == 2:
        return 6
    return 6*(n-1) - (n-2) + 4*( (n-1)*(n-2)/2 )

t = int(raw_input())

while (t > 0):
    t -= 1
    n = int(raw_input())
    print( present(n) )

